presto-iceberg/src/main/java/com/facebook/presto/iceberg/HiveTableOperations.java

```diff
@@ -450,4 +455,39 @@ private static int parseVersion(String metadataLocation)
             return -1;
         }
     }
+
+    /**
+     * Deletes the oldest metadata files if {@link
+     * TableProperties#METADATA_DELETE_AFTER_COMMIT_ENABLED} is true.
+     *
+     * @param base table metadata on which previous versions were based
+     * @param metadata new table metadata with updated previous versions
+     */
+    private void deleteRemovedMetadataFiles(TableMetadata base, TableMetadata metadata)
+    {
+        if (base == null) {
+            return;
+        }
+
+        boolean deleteAfterCommit =
+                metadata.propertyAsBoolean(
+                        TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED,
+                        TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED_DEFAULT);
+
+        if (deleteAfterCommit) {
+            Set<MetadataLogEntry> removedPreviousMetadataFiles =
```

Comment:
nit: since these files have not been removed yet having `removed` in the variable name is confusing. Maybe rename to `metadataFilesToRemove`?